packages/core-components/api-report.md

```diff
@@ -333,15 +333,32 @@ export type EmptyStateClassKey = 'root' | 'action' | 'imageContainer';
 // @public (undocumented)
 export type EmptyStateImageClassKey = 'generalImg';
 
-// Warning: (ae-forgotten-export) The symbol "State" needs to be exported by the entry point index.d.ts
+// Warning: (ae-forgotten-export) The symbol "ErrorBoundaryState" needs to be exported by the entry point index.d.ts
```

Comment:
Can you export this type too to get rid of the warning?

Reply:
Or rather - you know what, would it instead be better to _not_ have the state visible? Outside consumers should be shielded from internal implementation details. You'll see that your change here significantly increased the visible surface of the component. It might be best to do

```ts
export const ErrorBoundary: ComponentType<ErrorBoundaryProps>
  = class ErrorBoundary extends Component<ErrorBoundaryProps, State>
```

which means that seen from the outside, the exported symbol doesn't have a state parameter at all (and is `ComponentType` instead of `ComponentClass` which also is an implementation detail). But the class does have the state type, and now you can keep that purely private.